api/client/build.go

```diff
@@ -223,19 +234,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		Remove:         *rm,
 		ForceRemove:    *forceRm,
 		PullParent:     *pull,
-		Isolation:      *isolation,
 		CPUSetCPUs:     *flCPUSetCpus,
 		CPUSetMems:     *flCPUSetMems,
 		CPUShares:      *flCPUShares,
 		CPUQuota:       *flCPUQuota,
 		CPUPeriod:      *flCPUPeriod,
 		CgroupParent:   *flCgroupParent,
-		ShmSize:        *flShmSize,
 		Dockerfile:     relDockerfile,
+		ShmSize:        shmSize,
 		Ulimits:        flUlimits.GetList(),
-		BuildArgs:      flBuildArg.GetAll(),
 		AuthConfigs:    cli.configFile.AuthConfigs,
 	}
+	options.IsolationLevel = container.IsolationLevel(*isolation)
```

Comment:
Maybe my first comment was not clear enough, sorry:

```Go
options := types.ImageBuildOptions{
  // ...
  IsolationLevel: container.IsolationLevel(*isolation),
  BuildArgs: runconfig.ConvertKVStringsToMap(flBuildArg.GetAll()),
}
```

That's what I had in mind.